src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -3715,4 +3715,30 @@ BOOST_AUTO_TEST_CASE(AdaptiveMeanPoolingTestCase)
   BOOST_REQUIRE_EQUAL(arma::accu(delta), 1.5);
 }
 
+BOOST_AUTO_TEST_CASE(VAEModelTest)
+{
+  Sequential<>* decoder = new Sequential<>();
+
+    decoder->Add<Linear<>>(20, 10 * 10 * 24);
+    decoder->Add<LeakyReLU<>>();
+
+    decoder->Add<TransposedConvolution<>>(
+        24,  // Number of input activation maps.
+        16,  // Number of output activation maps.
+        5,   // Filter width.
+        5,   // Filter height.
+        1,   // Stride along width.
+        1,   // Stride along height.
+        0,   // Padding width.
+        0,   // Padding height.
+        10,  // Input width.
+        10); // Input height.
+
+    decoder->Add<LeakyReLU<>>();
+    decoder->Add<TransposedConvolution<>>(16, 1, 15, 15,
+        1, 1, 1, 1, 14, 14);
+
+    delete decoder;
```

Comment:
The indentation is off here, looks like we can remove two spaces.

Reply:
Hey @zoq, Let me know if this change makes sense or I'll try to fix the VAE model in some other way. I thought that visitor should set size of output width and output height but I think that happens during forward pass.